app/backend/src/couchers/interceptors.py

```diff
@@ -102,7 +103,10 @@ def intercept_service(self, continuation, handler_call_details):
             user_id = None
         else:
             # a valid user session was found
-            user_id, is_jailed = res
+            user_id, is_jailed, is_superuser = res
+
+            if auth_level == annotations_pb2.AUTH_LEVEL_ADMIN and not is_superuser:
+                return unauthenticated_handler("Permission denied", grpc.StatusCode.PERMISSION_DENIED)
```

Comment:
Why did you change this to `PERMISSION_DENIED`? I think it's good to keep the interceptor-level status code the same. It's a good idea to give it different text though, maybe "You need to be a superuser to perform that action".

`PERMISSION_DENIED` is often returned by the underlying API. I know in this case the waters get muddled a bit; but I think it's still better to use `UNAUTHENTICATED` iff it's from the interceptor.

Reply:
Doesn't `UNAUTHENTICATED` cause the frontend to log out?

Reply:
Or redirect you to the jail page. It depends on the error message.

I'm not sure where we're putting these sudo functions in the frontend: maybe the auth state should track that as another state in addition to jailed so it can show an extra "sudo" tab or similar? Then it'd be good to have the (frontend) interceptor catch that and un-sudo you?

Reply:
Using UNAUTHENTICATED when you are authenticated, but do not have permissions do to a thing seems very confusing. I would rather move the whole thing to a different interceptor then to return a status code that is functionally incorrect.

Reply:
Okay; leave as is!